gapic-common/lib/gapic/rest/threaded_fiber_enumerator.rb

```diff
@@ -0,0 +1,64 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Gapic
+  module Rest
+    ##
+    # Wraps a fiber-block (a block that contains Fiber.yield) of code into
+    # a Fiber that is wrapped into it's own thread and presents an iterable
+    # interface to it.
+    #
+    class ThreadedFiberEnumerator
+      FIBER_ERROR_SIGNAL = :fiber_error
```

Comment:
Seems like, semantically, this is actually "no more data" (and it just happens to raise FiberError because the fiber block has ended). I'd name the constant accordingly to make it obvious what's going on in the logic.

Reply:
Removed this file.